File: niro-server/niro-web/src/main/java/com/niro/web/enums/BuffAccountStatusEnum.java
package com.niro.web.enums;

import com.baomidou.mybatisplus.annotation.EnumValue;
import com.fasterxml.jackson.annotation.JsonValue;
import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * BUFF账号状态枚举
 *
 * @author niro
 * @since 2026-01-15
 */
@Getter
@AllArgsConstructor
public enum BuffAccountStatusEnum {

    /**
     * 正常
     */
    NORMAL("NORMAL", "正常"),

    /**
     * 封禁
     */
    BANNED("BANNED", "封禁"),

    /**
     * 市场访问限制
     */
    MARKET_RESTRICTED("MARKET_RESTRICTED", "市场访问限制"),

    /**
     * 下单限制
     */
    TRADE_RESTRICTED("TRADE_RESTRICTED", "下单限制"),

    /**
     * Cookie失效
     */
    INVALID("INVALID", "失效");

    @EnumValue
    @JsonValue
    private final String code;

    private final String description;
}

File: niro-client/src/enums/BuffAccountStatusEnum.ts
export enum BuffAccountStatusEnum {
  NORMAL = "NORMAL",
  BANNED = "BANNED",
  MARKET_RESTRICTED = "MARKET_RESTRICTED",
  TRADE_RESTRICTED = "TRADE_RESTRICTED",
  INVALID = "INVALID",
  COOLING_DOWN = "COOLING_DOWN",
  SCANNING = "SCANNING",
}

export const BuffAccountStatusMap = {
  [BuffAccountStatusEnum.NORMAL]: { label: "在线", theme: "success" },
  [BuffAccountStatusEnum.BANNED]: { label: "封禁", theme: "danger" },
  [BuffAccountStatusEnum.MARKET_RESTRICTED]: { label: "市场受限", theme: "warning" },
  [BuffAccountStatusEnum.TRADE_RESTRICTED]: { label: "交易受限", theme: "warning" },
  [BuffAccountStatusEnum.INVALID]: { label: "失效", theme: "danger" },
  [BuffAccountStatusEnum.COOLING_DOWN]: { label: "冷却中", theme: "warning" },
  [BuffAccountStatusEnum.SCANNING]: { label: "扫描中", theme: "primary" },
};
